src/SelfService/Infrastructure/Api/Prometheus/PrometheusClient.cs

```diff
@@ -0,0 +1,61 @@
+using System.Net.Http;
+using System.Text.Json;
+using SelfService.Domain.Exceptions;
+using SelfService.Application;
+
+namespace SelfService.Infrastructure.Api.Prometheus;
+
+
+public class PrometheusClient: IKafkaTopicConsumerService
+{
+
+    private readonly ILogger<PrometheusClient> _logger;
+    private readonly HttpClient _httpClient;
+
+    public PrometheusClient(ILogger<PrometheusClient> logger, HttpClient httpClient)
+    {
+        _logger = logger;
+        _httpClient = httpClient;
+    }
+
+    private IEnumerable<string> GetConsumersFromResponse(Response? response, string topic)
+    {
+        List<string> consumers = new List<string>();
+        if (response == null || response.data == null || response.data.result == null) {
+            return new List<string>();
+        }
+        foreach (Result result in response.data.result) {
+            if (result.metric == null || result.metric.topic == null || result.metric.consumergroup == null) {
+                continue;
+            }
+            if (result.metric.topic == topic)
+            {
+                consumers.Add(result.metric.consumergroup);
+            }
```

Comment:
Will they ever be null? In case they are, do we want a log if it is an unusual thing to happen?